### ISSUE TITLE
Passing a version_id to generate_blob_sas creates a SAS

### ISSUE THREAD

**Issue by piercsi** [Azure/azure-sdk-for-python#28545]
It should be made clear in the documentation that when using generate_blob_sas to fetch specific versions of a document, the returned SAS is not valid unless the version id is appended to the resulting query string in the format:

`sas_that_works = f'{returned_sas}&versionid={versionid}'`

where `versionid` exactly matches the created date/time of the blob in YYYY-MM-DDTHH:MM:SS.NNNNNNNZ format.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: bc13eca7-8c35-456c-cb38-016d2fa40df5
* Version Independent ID: d89dc470-199e-fe2a-e8ba-ec35c73e1bc4
* Content: [azure.storage.blob package](https://learn.microsoft.com/en-us/python/api/azure-storage-blob/azure.storage.blob?view=azure-python)
* Content Source: [docs-ref-autogen/azure-storage-blob/azure.storage.blob.yml](https://github.com/MicrosoftDocs/azure-docs-sdk-python/blob/main/docs-ref-autogen/azure-storage-blob/azure.storage.blob.yml)
* Service: **storage**
* GitHub Login: @lmazuel
* Microsoft Alias: **lmazuel**

**Comment by kashifkhan** [comment]
Thanks for the feedback @piercsi . We will investigate and get back to you asap.

**Comment by jalauzon-msft** [comment]
Hi @piercsi Simon, thanks for the request. While it is true that a SAS generated with a `version_id` can only be used on that specific version of the blob, I'm not sure documenting this on the result of the `generate_blob_sas` is the right thing to do. The resulting SAS token is intended to be used with an API in the SDK where you would already have to specify the `version_id` as an argument to that operation. For example:

```py
blob_client = BlobClient(..., credential=<sas_token>)
props = blob_client.get_blob_properties(version_id='YYYY-MM-DDTHH:MM:SS.NNNNNNNZ')
```

In this case, you would not want to add the `versionid` query param to your SAS token as `get_blob_properties` will be adding it to the API call for you and you'd actually end up with two `versionid` which would be invalid. So ideally, when using the SDK, you should never have to add the `versionid` query parameter yourself even if you have a SAS for a version.

**Comment by piercsi** [comment]
I don't disagree with the rationale for how it works; my suggestion is that it could be explained better in the documentation somewhere.

The confusion (in opinion) is that when using a SAS, generate_blob_sas creates all the query string parameters for you, except the `versionid`, even when you supply a `version_id` parameter. It feels counterintuitive.

**Comment by jalauzon-msft** [comment]
Hi @piercsi Simon, sorry for the delay but we've taken your feedback and merged #29802 to update the documentation to include:

>Note that the 'versionid' query parameter is not included in the output SAS. Therefore, please provide the 'version_id' parameter to any APIs when using the output SAS to operate on a specific version.

The online documentation is auto-generated from the code and will update within the next couple of weeks. Thanks!
